### PR TITLE
fix: add service account names to values

### PR DESCRIPTION
**Pull request by jaygridley**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -141,52 +180,49 @@ variable "argo_info" {
...
-  #   "annotations" : {}
-  #   "labels" : {}
-  #   "finalizers" : [
-  #     "resources-finalizer.argocd.argoproj.io"
-  #   ]
-  # }
+  type = any
+  default = {
+    "finalizers" : [
+      "resources-finalizer.argocd.argoproj.io"
```
Comment (JaroVojtek) [inline variables.tf:192]: This finalizer we dont want as default especially for argocd as I understood from @tomas-balaz post

**Review by JaroVojtek — COMMENTED** [review]
@martinhaus @dojci can check as well If I not forgot on something during review. Thansk

File: variables.tf [hunk 2]
```diff
@@ -37,35 +33,64 @@ variable "helm_release_name" {
...
+  description = "Whether to create and use RBAC resources"
+}
+
+variable "service_account_create_server" {
+  type        = bool
+  default     = true
+  description = "Whether to create Service Account for the server"
 }
 
-variable "self_managed" {
```
[the remark this answers is not in the document]
Comment (jaygridley) [reply variables.tf:65]: my bad during merge

File: variables.tf [hunk 3]
```diff
@@ -82,8 +125,8 @@ variable "irsa_additional_policies" {
 
 variable "irsa_role_name_prefix" {
   type        = string
-  default     = "argocd-irsa"
-  description = "The IRSA role name prefix for vector"
+  default     = "<$addon-name>-irsa"
```
[the remark this answers is not in the document]
Comment (jaygridley) [reply variables.tf]: my bad during merge

File: iam.tf [hunk 4]
```diff
@@ -1,47 +1,57 @@
...
 
-    condition {
-      test     = "StringEquals"
-      variable = "${replace(var.cluster_identity_oidc_issuer, "https://", "")}:sub"
 
-      values = [
-        "system:serviceaccount:${var.namespace}:${var.service_account_name_application_controller}"
-      ]
-    }
+  dynamic statement {
```
[the remark this answers is not in the document]
Comment (jaygridley) [reply iam.tf]: nice idea, I like it, reworked according to the example

File: variables.tf (hunk 1 above)
Comment (jaygridley) [reply variables.tf]: my bad during merge

File: examples/basic/providers.tf [hunk 5]
```diff
@@ -10,16 +10,10 @@ data "aws_eks_cluster_auth" "this" {
   name = module.eks_cluster.eks_cluster_id
 }
 
-provider "kubernetes" {
```
Comment (JaroVojtek) [inline examples/basic/providers.tf:13]: Dont we need kubernetes provider ? in the examples we are calling module with `argo_helm_enabled = false`. In that case we use kubernetes provider for installation. If we remove this provider from `examples/basic/providers.tf` then `module "argocd_self_managed_kubernetes"` will fail.

**Review by JaroVojtek — CHANGES_REQUESTED** [review]

File: examples/basic/providers.tf (hunk 5 above)
Comment (jaygridley) [reply examples/basic/providers.tf:13]: It is not possible to use kubernetes_manifest without an existing cluster so this installation method will fail anyway. I've update docs to make this more obvious and put the provider back for the case when the cluster is up and running.

**Review by JaroVojtek — APPROVED** [review]

**Comment by JaroVojtek** [comment]
LGTM

**Pull request merged**